ingredient/views.py

```diff
@@ -1,3 +1,27 @@
-from django.shortcuts import render
+from django.core.exceptions import ObjectDoesNotExist
+from django.http            import JsonResponse
+from django.views           import View
+from ingredient.models      import Ingredient
 
-# Create your views here.
+
+class IngredientView(View):
+    def get (self, request, ingredient_id): 
+        try: 
+            ingredient  = Ingredient.objects.get(id=ingredient_id)
+            recipe_all  = ingredient.recipe.all()
+            recipe_list = [{
+                "id"        : recipe.id,
+                "name"      : recipe.name,
+                "image_url" : recipe.image_url} for recipe in recipe_all]
+            result = {
+                "id"               : ingredient.id,
+                "name"             : ingredient.name,
+                "price"            : ingredient.price,
+                "info"             : ingredient.info,
+                "storage"          : ingredient.storage,
+                "image_url"        : ingredient.image_url,
+                "related_recipe"   : recipe_list
+            }
+            return JsonResponse({"ingredient":result})
+        except ObjectDoesNotExist:
```

Comment:
아래와 같이 어떤 객체가 존재하지 않는지 좀더 명확하게 에러를 잡아줄 수 있을 것 같습니다.
`ObjectDoesNotXist` class를 한번 더 감싼 `Ingredient.DoesNotExist`를 사용하면 디버깅하기도 더 쉽고 코드 가독성도 좋아집니다!

![image](https://user-images.githubusercontent.com/53214510/122143815-2c11b280-ce8d-11eb-8738-1c3fa19108eb.png)


```suggestion
        except Ingredient.DoesNotExist:
```

Reply:
장고에서 더 지원해주는 기능이었군요!! 이해안갔었는데 짚어주셔서 이해됐어요!